ci/constraints-test/py312-min-install.txt

```diff
@@ -0,0 +1,3 @@
+numpy==1.26.0
+pandas==2.2.0
+sqlalchemy==2.0.0 # pandas 2 requires sqlalchemy2
```

Comment:
Technically `1.4.49` works with Python 3.12 but because the min version of pandas for `3.12` is `2.2.0` we have to use SQLAlchemy 2.0.
https://www.sqlalchemy.org/blog/2023/07/05/sqlalchemy-1.4.49-released/